<?xml version="1.0"?>
<launch>
    <!-- spawn joy node -->
    <!-- <node pkg="joy" 
          type="joy_node" 
          name="joy_node" /> -->

	<node pkg="sbus_serial" type="sbus_serial_node" name="sbus_node" output="screen">
		<param name="port" value="/dev/sbus" />
		<param name="refresh_rate_hz" value="25" />
		<param name="rxMinValue" value="695" />
		<param name="rxMaxValue" value="1509" />
		<param name="outMinValue" value="0" />
		<param name="outMaxValue" value="1000" />
		<param name="silentOnFailsafe" value="false" />

		<!-- To only publish messages when a specific channel is within a certain range, use the following three parameters -->
		<!--
		The enableChannelNum is the channel number to check. Note, this is the channel number, not the array index. So channel 1 is "1" and not "0".
		If enableChannelNum is set to -1 or 0 or higher than 16, no check is performed.
		Set enableChannelProportionalMin and enableChannelProportionalMax to the "proportional values" within which publishing is enabled.
		Example: to publish only if channel 5 is above 50%:
			<param name="enableChannelNum" value="5" />
			<param name="enableChannelProportionalMin" type="double" value="0.5" />
			<param name="enableChannelProportionalMax" type="double" value="1.0" />
		-->
		<param name="enableChannelNum" value="-1" />
		<param name="enableChannelProportionalMin" type="double" value="0.0" />
		<param name="enableChannelProportionalMax" type="double" value="0.0" />

	</node>

    <!-- spawn controller node -->
    <node pkg="pro_lower_controller" type="pro_lower_platform" name="pro_lower_platform" output="screen">
        <param name="canSeries"     type="string"    value="can0"/>
        <param name="speedMax"     type="double"    value="5.0"/>
        <param name="speedMin"     type="double"    value="-1.0"/>
    </node>

    <!-- RpLiadr s1 -->
    <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
        <param name="serial_port"         type="string" value="/dev/laser"/>
        <param name="serial_baudrate"     type="int"    value="256000"/>
        <param name="frame_id"            type="string" value="laser"/>
        <param name="inverted"            type="bool"   value="false"/>
        <param name="angle_compensate"    type="bool"   value="true"/>
    </node>

    <!-- PlotJuggle -->
    <node name="plotjuggler_with_layout" 
          pkg="plotjuggler"
          type="plotjuggler"
          args="--layout $(find pro_lower_controller)/rviz/plotjuggler.xml"/>
</launch>
